In [8]:
from googleapiclient import discovery
import json

API_KEY = 'AIzaSyBxCxXlppjig6vScoR1wJf09bWx4Grco3Q'

client = discovery.build(
  "commentanalyzer",
  "v1alpha1",
  developerKey=API_KEY,
  discoveryServiceUrl="https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
  static_discovery=False,
)

analyze_request = {
  'comment': { 'text': 'friendly greetings from python' },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
print(json.dumps(response, indent=2))

{
  "attributeScores": {
    "TOXICITY": {
      "spanScores": [
        {
          "begin": 0,
          "end": 30,
          "score": {
            "value": 0.02543884,
            "type": "PROBABILITY"
          }
        }
      ],
      "summaryScore": {
        "value": 0.02543884,
        "type": "PROBABILITY"
      }
    }
  },
  "languages": [
    "en"
  ],
  "detectedLanguages": [
    "en"
  ]
}


### My hypothesis: Perspective will be more likely to incorrectly classify non-toxic sentences containing certain controversial keywords as toxic, compared to similar sentences without those keywords. This hypothesis suggests that when non-toxic sentences include certain controversial words or topics, the Perspective model is more likely to mistakenly consider them as toxic, even though the sentences are not offensive or abusive. In simpler terms, the presence of these controversial keywords might confuse the model and lead to incorrect toxic classifications.

### By controversial terms this may include: 
### 1. Religion-related keywords: Christianity, Islam, Hinduism, Atheism, god, jews 
### 2. Political keywords: Democrat, Republican, Socialism, Capitalism, Communism
### 3. Social issues keywords: Abortion, Immigration, Gun control, LGBTQ+, Feminism

### I will be adding/editing the words in the sentences to test the hypothesis 

In [27]:
# Test 1
test = "Dear god this site is horrible" # not toxic

analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}

response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 1: " + str(toxicityProbability))

# Test 2
test = "Islam and Slavery   Is WikipediaArticles for deletionIslam and Slavery  Would you care to vote Thx" # not toxic 

analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 2: " + str(toxicityProbability))

# Test 3
test = "Conservative Democratic Alliance   I have reverted the unsubstantiated rubbish you wrote about CDA Since I am Chairman I know somewhat more about my organisation than you do  Why are you interested in my group" # not toxic 


analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 3: " + str(toxicityProbability))

# Test 4
test = "This message is regarding the page Abortion Please stop  If you continue to vandalize pages you will be blocked from editing Wikipedia" # not toxic 

analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 4: " + str(toxicityProbability))

# Test 5
test = "That is of course wrong Academic books can be better than articles since thay can list their sources in great detail For example the Black Book of Communism written by several academics lists thousands of sources" # not toxic
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 5: " + str(toxicityProbability))

# Test 6
test = "jimmy malone is scottish hes a fictional character and david mamet say he was from scotland in his script notes and RepublicanJacobite is clearly a troll and racist so people are scots irish" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 6: " + str(toxicityProbability))


# Test 7
test = "REDIRECT TalkDepartment of Immigration and Multicultural Affairs 2006–07" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 7: " + str(toxicityProbability))


# Test 8 
test = "Its not a theory its fact supported by numerous RS And it constitutes an instance of gun control so yes it is relevant to an article on gun control" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 8: " + str(toxicityProbability))


# Test 9
test = "lol stupid Muslims burned their own shit" # It is toxic
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 9: " + str(toxicityProbability))



# Test 10 
test = "You clearly dont know what socialism is You seem to confuse socialism with communism which are connected but not the same" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 10: " + str(toxicityProbability))



# Test 11
test = "Update Notice  There are according to the reddit itself roughly 590000 redditors subscribing to the Atheism reddit almost double the number listed FYI" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 11: " + str(toxicityProbability))



# Test 12
test = "Sexuality of Adolf Hitler" # not toxic 
analyze_request = {
  'comment': { 'text': test },
  'requestedAttributes': {'TOXICITY': {}}
}


response = client.comments().analyze(body=analyze_request).execute()
toxicityProbability = response['attributeScores']['TOXICITY']['summaryScore']['value']
print("Test 12: " + str(toxicityProbability))

Test 1: 0.4014846
Test 2: 0.37751234
Test 3: 0.3811502
Test 4: 0.20172562
Test 5: 0.0665887
Test 6: 0.509388
Test 7: 0.010681152
Test 8: 0.038991302
Test 9: 0.96069145
Test 10: 0.13214645
Test 11: 0.038991302
Test 12: 0.39842087


### Test 1: toxicity prob = 0.4 correctly determining this sentence is not toxic. 
### Test 2: toxicity prob = 0.37 correctly determining this sentence is not toxic. 
### Test 3: toxicity prob = 0.38 correctly determining this sentence is not toxic. 
### Test 4: toxicity prob = 0.2 correctly determining this sentence is not toxic. 
### Test 5 :toxicity prob = 0.06 correctly determining this sentence is not toxic.
### Test 6 :toxicity prob = 0.5 incorrectly determining this sentence is toxic.
### Test 7 :toxicity prob = 0.01 correctly determining this sentence is not toxic.
### Test 8: toxicity prob = 0.03 correctly determining this sentence is not toxic. 
### Test 9: toxicity prob = 0.96 incorrectly determining this sentence is toxic. 
### Test 10: toxicity prob = 0.13 correctly determining this sentence is not toxic. 
### Test 11: toxicity prob = 0.03 correctly determining this sentence is not toxic. 
### Test 12: toxicity prob = 0.39 correctly determining this sentence is not toxic. 



### After running these test results I have determined that my hypothesis was incorrect because on average the tests correctly determined the toxicity probabality of sentences with controversial terms. Furthermore, there were two tests that incorrectly determined the toxicity probability which tells us that there is a small bias.


### In conclusion, there is a small bias within my hypothesis after running 12 tests. 










